Source/WebKit/NetworkProcess/NetworkConnectionToWebProcess.cpp

```diff
@@ -1414,35 +1414,16 @@ void NetworkConnectionToWebProcess::installMockContentFilter(WebCore::MockConten
 }
 #endif
 
-
 #if ENABLE(LOGD_BLOCKING_IN_WEBCONTENT)
-static CString validatedASCIIString(const String& string, unsigned maxLength)
-{
-    String shortString = string.substring(0, maxLength);
-    CString validatedString = shortString.ascii();
-    auto stringLength = validatedString.length();
-
-    for (unsigned i = 0; i < stringLength; i++) {
-        char c = validatedString.data()[i];
-        if (c >= 32 && c <= 126)
-            continue;
-        validatedString.mutableData()[i] = ' ';
-    }
-    return validatedString;
-}
-
-void NetworkConnectionToWebProcess::logOnBehalfOfWebContent(const String& logChannel, const String& logCategory, const String& logString, uint8_t logType, int32_t pid)
+void NetworkConnectionToWebProcess::logOnBehalfOfWebContent(Vector<char>&& logChannel, Vector<char>&& logCategory, IPC::DataReference&& logString, uint8_t logType, int32_t pid)
```

Comment:
Any reason we're using an IPC::DataReference for the logString but not for the other 2 strings? Looks to me that we don't need to memcpy those characters to vectors just to log them?

Reply:
Yes, we need to copy the log channel and category strings, since they will be freed after returning from the hook. For the log string, we own it's memory, so we can use an IPC::DataReference there.

Reply:
Sorry, I don't understand this comment. This is coming from IPC so we own the memory no matter what. The question is, do we need to memcpy the characters to a Vector before logging or can we log straight from a IPC::DataReference that points to the IPC memory.

Looking at the code, I couldn't see a good reason for using Vectors here.

Reply:
> Sorry, I don't understand this comment. This is coming from IPC so we own the memory no matter what. The question is, do we need to memcpy the characters to a Vector before logging or can we log straight from a IPC::DataReference that points to the IPC memory.
> 
> Looking at the code, I couldn't see a good reason for using Vectors here.

Ah, I see what you mean. I have changed to IPC::DataReference parameters in the latest patch.

Thanks for reviewing!